model-optimizer/unit_tests/extensions/ops/select_test.py

```diff
@@ -79,6 +79,56 @@ def test_select_infer_condition_false(self):
+                                                                ('select_output', {'shape': np.array([2, 2]),
+                                                                                   'value': np.zeros((2, 2))}),
+                                                                ('then_data', {'value': None, 'shape': np.array([2, 2])})
+                                                                ])
+
+        # We should propagate shapes and values
+        graph_ref = build_graph_with_attrs(nodes_with_attrs=self.nodes,
+                                           edges_with_attrs=self.edges,
+                                           update_nodes_attributes=[('select_output', {'shape': np.array([2, 2]),
+                                                                                       'value': np.zeros((2, 2))}),
+                                                                    ('then_data',
+                                                                     {'value': None, 'shape': np.array([2, 2])})])
+        node = Node(graph, 'select')
+        node.infer(node)
+        (flag, resp) = compare_graphs(graph, graph_ref, 'select_output', check_op_attrs=True)
+        self.assertTrue(flag, resp)
+
+    # if one of the branches is None then np.where shouldn't be used to avoid object dtype in output
+    # res = np.where(condition, numpy_array_of_int[float]_dtype, None)
+    # print(res.dtype) => object which is not compatible with other numeric dtypes, will fail further without
+    # clear explanation, need to catch such cases as soon as possible
+    def test_select_infer_None_else_branch(self):
+        graph = build_graph_with_attrs(nodes_with_attrs=self.nodes, edges_with_attrs=self.edges,
+                                       update_nodes_attributes=[('condition_data', {'value': np.array([True])}),
+                                                                ('select_output', {'shape': np.array([2, 2]),
+                                                                                   'value': np.ones((2, 2))}),
+                                                                (
+                                                                'else_data', {'value': None, 'shape': np.array([2, 2])})
+                                                                ])
+
+        # We should propagate shapes and values
+        graph_ref = build_graph_with_attrs(nodes_with_attrs=self.nodes,
+                                           edges_with_attrs=self.edges,
+                                           update_nodes_attributes=[('select_output', {'shape': np.array([2, 2]),
+                                                                                       'value': np.ones((2, 2))}),
+                                                                    ('else_data',
+                                                                     {'value': None, 'shape': np.array([2, 2])})])
```

Comment:
the same

Reply:
have completely rewritten these tests, and types now are defined explicitly